ibis/expr/types/arrays.py

```diff
@@ -471,6 +471,294 @@ def filter(self, predicate: Callable[[ir.Value], ir.BooleanValue]) -> ir.ArrayVa
+        │ [2, 2]               │
+        │ NULL                 │
+        └──────────────────────┘
+        >>> t.arr.remove(2)
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ ArrayRemove(arr, 2)  ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ array<int8>          │
+        ├──────────────────────┤
+        │ [3]                  │
+        │ []                   │
+        │ [42]                 │
+        │ []                   │
+        │ NULL                 │
+        └──────────────────────┘
+        """
+        return ops.ArrayRemove(self, other).to_expr()
+
+    def unique(self) -> ir.ArrayValue:
+        """Return the unique values in an array.
+
+        !!! note "Element ordering in array may not be retained."
+
+        Returns
+        -------
+        ArrayValue
+            Unique values in an array
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [[1, 3, 3], [], [42, 42], None]})
+        >>> t
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ arr                  ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ array<int8>          │
+        ├──────────────────────┤
+        │ [1, 3, ... +1]       │
+        │ []                   │
+        │ [42, 42]             │
+        │ NULL                 │
+        └──────────────────────┘
+        >>> t.arr.unique()
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ ArrayDistinct(arr)   ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ array<int8>          │
+        ├──────────────────────┤
+        │ [3, 1]               │
+        │ []                   │
+        │ [42]                 │
+        │ NULL                 │
+        └──────────────────────┘
+        """
+        return ops.ArrayDistinct(self).to_expr()
+
+    def sort(self) -> ir.ArrayValue:
+        """Sort the elements in an array.
+
+        Returns
+        -------
+        ArrayValue
+            Sorted values in an array
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [[3, 2], [], [42, 42], None]})
+        >>> t
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ arr                  ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ array<int8>          │
+        ├──────────────────────┤
+        │ [3, 2]               │
+        │ []                   │
+        │ [42, 42]             │
+        │ NULL                 │
+        └──────────────────────┘
+        >>> t.arr.sort()
+        ┏━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ ArraySort(arr)       ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━┩
+        │ array<int8>          │
+        ├──────────────────────┤
+        │ [2, 3]               │
+        │ []                   │
+        │ [42, 42]             │
+        │ NULL                 │
+        └──────────────────────┘
+        """
+        return ops.ArraySort(self).to_expr()
+
+    def union(self, other: ir.ArrayValue) -> ir.ArrayValue:
```

Comment:
I wonder what should we do with the set datatype

Reply:
Maybe we should remove it?